compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -1919,6 +1919,12 @@ private BType computeIntersectionType(BLangIntersectionTypeNode intersectionType
             isErrorIntersection = true;
         }
 
+        if ((typeOne.tag == typeTwo.tag && typeOne.tag != TypeTags.ERROR)
+            || (!hasReadOnlyType && (typeOne.tag != TypeTags.ERROR || typeTwo.tag != TypeTags.ERROR))) {
```

Comment:
Can't we use `!isErrorIntersection && !hasReadOnlyType && (typeOne.tag != TypeTags.ERROR || typeTwo.tag != TypeTags.ERROR)` instead?

Reply:
Also, shouldn't this result in one of two errors? One for intersections that are not yet supported vs one for invalid intersection types (DiagnosticErrorCode.INVALID_INTERSECTION_TYPE)?

`int & string x = 1;` also results in `unsupported intersection 'int & string'` with these changes rather than `invalid intersection type 'int & string': no intersection`.

Reply:
```ballerina
type A int & int;
```
This crash for me in SL Alpha-5

Reply:
Shall we create an issue? We need to fix the error message for Swan Lake anyway, right?

Reply:
Issue created https://github.com/ballerina-platform/ballerina-lang/issues/31101